builtin/mainmenu/init.lua

```diff
@@ -77,13 +77,28 @@ local function init_globals()
 
 	if PLATFORM == "Android" then
 		local world_list = core.get_worlds()
-		local world_index = table.indexof(world_list, "singleplayerworld")
-
-		if world_index == -1 then
+		local world_index
+
+		local found_singleplayerworld = false
+		for i, v in ipairs(world_list) do
+			if world.name == "singleplayerworld" then
```

Comment:
It reads a bit funny, he only wants the first entry in the list.

You can write it a lot more readable like this:

```
if world.name == "singleplayerworld" then
        world_index = next(world_list, nil)
        found_singleplayerworld = true
end
```